### ISSUE TITLE
Crash when trying to run

### ISSUE THREAD

**Issue by jpschewe** [github/git-sizer#11]
Running against a checkout of git@github.com:jpschewe/fll-sw.git. 
```
>uname -a
Linux jon-laptop 4.13.0-36-generic #40-Ubuntu SMP Fri Feb 16 20:07:48 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

>git-sizer  
Processing blobs: 22223                                                                   Processing trees: 24428                      
Processing commits: 5194                     
Matching commits to trees: 5194                                                           Processing annotated tags: 94
Processing references: 156
panic: 8 tag records remain!                                                                                    
goroutine 1 [running]:
github.com/github/git-sizer/sizes.(*Graph).HistorySize(0xc420001800, 0x0, 0x0, 0x0, 0x0, 0
x0, 0x0, 0x0, 0x0, 0x0, ...)                                                              
        /home/mhagger/github/proj/git-sizer/git/.gopath/src/github.com/github/git-sizer/si
zes/graph.go:398 +0x2c9
github.com/github/git-sizer/sizes.ScanRepositoryUsingGraph(0xc420010510, 0x5687c0, 0x2, 0x
1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)                                                     
        /home/mhagger/github/proj/git-sizer/git/.gopath/src/github.com/github/git-sizer/si
zes/graph.go:335 +0x1df6                                                                  main.mainImplementation(0x0, 0x0)
        /home/mhagger/github/proj/git-sizer/git/.gopath/src/github.com/github/git-sizer/git-sizer.go:150 +0x8ef 
main.main()                                                                                       /home/mhagger/github/proj/git-sizer/git/.gopath/src/github.com/github/git-sizer/gi
t-sizer.go:46 +0x26                                 
```

**Comment by Abdulmanahtalib** [comment]
Tq

**Comment by larsxschneider** [comment]
I did run into the same problem with multiple repos but I haven't see a pattern amongst them yet. Looks like [`finalizeTagSize()`](https://github.com/github/git-sizer/blob/6cda80729b146d3cdc8b0df7db2bb6938621940b/sizes/graph.go#L721) is not called in some cases (I am not familiar with the code base)?!

**Comment by mhagger** [comment]
My first guess is that this happens if one annotated tag points at another one and git outputs them in an unexpected order. I'm travelling this week but I'll try to look into it early next week.

As a temporary workaround, you can process most of your repository's history successfully if you add the `--branches` option to the `git-sizer` command line.

**Comment by jpschewe** [comment]
Thank you, that flag lets it run and give me some results.

**Comment by opragel** [comment]
I'm experiencing the same issue for what looks like the same reason possibly. Running --branches allows me to get some output.

```
~/bin/git-sizer --tags                                                                                                                                         [git:master] ✔ []
Processing blobs: 116410                        
Processing trees: 360387                        
Processing commits: 56035                        
Matching commits to trees: 56035                        
Processing annotated tags: 542                        
Processing references: 623                        
panic: 3 tag records remain!

goroutine 1 [running]:
github.com/github/git-sizer/sizes.(*Graph).HistorySize(0xc420001980, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/mhagger/github/proj/git-sizer/git/.gopath/src/github.com/github/git-sizer/sizes/graph.go:398 +0x2c9
github.com/github/git-sizer/sizes.ScanRepositoryUsingGraph(0xc4200425c0, 0xc42000a0c0, 0x2, 0xc42000a001, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/mhagger/github/proj/git-sizer/git/.gopath/src/github.com/github/git-sizer/sizes/graph.go:335 +0x1df6
main.mainImplementation(0x0, 0x0)
	/home/mhagger/github/proj/git-sizer/git/.gopath/src/github.com/github/git-sizer/git-sizer.go:150 +0x8ef
main.main()
	/home/mhagger/github/proj/git-sizer/git/.gopath/src/github.com/github/git-sizer/git-sizer.go:46 +0x26
```

**Comment by Abdulmanahtalib** [comment]
My problem

**Comment by Abdulmanahtalib** [comment]
Run

**Comment by 0xmohit** [comment]
This error is also encountered when running against [`rust`](https://github.com/rust-lang/rust) repo.

**Comment by mhagger** [comment]
I think that this issue should be fixed by #21. Anybody who doesn't mind building from source can try it out, though I'd like to add some automated tests before actually merging and creating a new release.

**Comment by 0xmohit** [comment]
@mhagger The issue appears to be fixed by #21.

**Comment by opragel** [comment]
@0xmohit #21 fixed the issue for me. Thanks!
